src/Dfc.CourseDirectory.WebV2/Behaviors/ServiceCollectionExtensions.cs

```diff
@@ -1,23 +1,53 @@
 ﻿using System;
 using System.Linq;
+using System.Reflection;
 using MediatR;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace Dfc.CourseDirectory.WebV2.Behaviors
 {
     public static class ServiceCollectionExtensions
     {
-        public static IServiceCollection AddBehaviors(this IServiceCollection services)
+        public static IServiceCollection AddBehaviors(
+            this IServiceCollection services,
+            Assembly handlersAssembly)
         {
-            foreach (var type in typeof(ServiceCollectionExtensions).Assembly.GetTypes())
+            foreach (var type in handlersAssembly.GetTypes())
             {
+                RegisterRequireUserCanAccessCourseBehavior(type);
                 RegisterRequireUserCanSubmitQASubmissionBehavior(type);
                 RegisterRestrictProviderTypeBehavior(type);
                 RegisterRestrictQAStatusBehavior(type);
             }
 
             return services;
 
+            void RegisterRequireUserCanAccessCourseBehavior(Type type)
```

Comment:
This junk is here due to limitations with built-in IOC container (which is being fixed for .NET 5).